### ISSUE TITLE
AzureAD/EntraID Authentication for database backend 

### ISSUE THREAD

**Issue by nareshbogathi** [DefectDojo/django-DefectDojo#10815]
Is your feature request related to a problem? Please describe
As a DevSecOps engineer I am required by security policy to authenticate database connections using Active Directory / EntraID instead of native postgresql user and password.

Describe the solution you'd like
The possibility to use Azure AD token based authentication for database connectivity

Describe alternatives you've considered
None. This is a new company policy.

Additional context
https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-azure-ad-authentication

https://stackoverflow.com/questions/77135512/sqlalchemy-connect-to-azure-postgresql-flexible-database-with-token-authenticati

**Comment by twwd** [comment]
Seems to be a one-on-one duplicate of #9610

**Comment by nareshbogathi** [comment]
I wanted to check if DefectDojo currently supports connecting to PostgreSQL using Azure Entra (formerly Azure AD) tokens instead of traditional username and password authentication.

If this is not supported yet, are there any plans to add this functionality in the future? This feature would align well with enhanced security practices by enabling passwordless and token-based authentication.

Looking forward to your response.

**Comment by nareshbogathi** [comment]
@mtesauro @twwd 
I wanted to revisit this topic as we’re currently blocked by an internal compliance requirement.
We have an Azure policy (AAB Database for PostgreSQL Flexible Server - Entra ID Password Authentication v2) that enforces the use of Entra ID–based authentication instead of username/password. Our PostgreSQL flexible server is flagged as non-compliant until this is resolved.

Because of this, we’re unable to use standard credentials for DefectDojo database connectivity.
Has there been any progress or reconsideration around supporting Azure Entra ID (AAD) / token-based authentication for PostgreSQL connections? Or is PgBouncer still the only recommended workaround in this scenario?

This is becoming a blocker for us to remain compliant while continuing to use DefectDojo, so any updates or guidance would be greatly appreciated.

Thanks!
